src/main/kotlin/prison/nomis.kt

```diff
@@ -4,83 +4,97 @@ import com.structurizr.model.Location
 import com.structurizr.model.Model
 import com.structurizr.model.Container
 import com.structurizr.model.SoftwareSystem
+import com.structurizr.view.ViewSet
 
+import uk.gov.justice.hmpps.architecture.HMPPSSoftwareSystem
 import uk.gov.justice.hmpps.architecture.shared.Tags
 
-class NOMIS(model: Model) {
-  val system: SoftwareSystem
-  val db: Container
-  val elite2api: Container
-
-  init {
-    val cloudPlatform = model.getDeploymentNodeWithName("Cloud Platform")
-    val sqs = cloudPlatform.getDeploymentNodeWithName("SQS")
-    val sns = cloudPlatform.getDeploymentNodeWithName("SNS")
-    val kubernetes = cloudPlatform.getDeploymentNodeWithName("Kubernetes")
-    val elasticSearch = cloudPlatform.getDeploymentNodeWithName("ElasticSearch")
-
-    system = model.addSoftwareSystem("NOMIS", """
-    National Offender Management Information System,
-    the case management system for offender data in use in custody - both public and private prisons
-    """.trimIndent()).apply {
-      setLocation(Location.Internal)
-    }
+class NOMIS private constructor() {
 
-    db = system.addContainer("NOMIS database", null, "Oracle").apply {
-      Tags.DATABASE.addTo(this)
-    }
+  companion object: HMPPSSoftwareSystem {
+    lateinit var system: SoftwareSystem
+    lateinit var db: Container
+    lateinit var elite2api: Container
 
-    system.addContainer("NOMIS Web Application",
-        "Web Application fronting the NOMIS DB - used by Digital Prison team applications and services",
-        "Weblogic App Sever").apply {
-      uses(db, "JDBC")
-    }
+    override fun defineModelEntities(model: Model) {
+      val cloudPlatform = model.getDeploymentNodeWithName("Cloud Platform")
+      val sqs = cloudPlatform.getDeploymentNodeWithName("SQS")
+      val sns = cloudPlatform.getDeploymentNodeWithName("SNS")
+      val kubernetes = cloudPlatform.getDeploymentNodeWithName("Kubernetes")
```

Comment:
Would it be possible to replace name-based lookups with `CloudPlatform.kubernetes`?

Reply:
:+1: I've updated this now